pkg/sentry/devices/nvproxy/frontend_unsafe.go

```diff
@@ -47,6 +47,29 @@ func rmControlInvoke[Params any](fi *frontendIoctlState, ioctlParams *nvgpu.NVOS
 	return n, nil
 }
 
+func ctrlAttachGpuInvoke(fi *frontendIoctlState, ioctlParams *nvgpu.NVOS54Parameters, ctrlParams *nvgpu.NV00FD_CTRL_ATTACH_GPU_PARAMS) (uintptr, error) {
+	origDevDescriptor := ctrlParams.DevDescriptor
+	DevDescriptor, _ := fi.t.FDTable().Get(int32(origDevDescriptor))
+	if DevDescriptor == nil {
+		return 0, linuxerr.EINVAL
+	}
+	DevDesc, ok := DevDescriptor.Impl().(*frontendFD)
+	if !ok {
+		return 0, linuxerr.EINVAL
+	}
+	ctrlParams.DevDescriptor = uint64(DevDesc.hostFD)
+	n, err := rmControlInvoke(fi, ioctlParams, ctrlParams)
+	ctrlParams.DevDescriptor = origDevDescriptor
+	if err != nil {
+		return n, err
+	}
+	// ctrlParams.DevDescriptor = origDevDescriptor
+	if _, err := ctrlParams.CopyOut(fi.t, addrFromP64(ioctlParams.Params)); err != nil {
```

Comment:
Do we technically need to copy out? Looking at the [driver code](https://github.com/NVIDIA/open-gpu-kernel-modules/blob/caa2dd11a0db42e5d6130f627dd15ef404b7597d/src/common/sdk/nvidia/inc/ctrl/ctrl00fd.h#L153), it seems like all the fields are only used for input.

Reply:
Oh right, so the command is all `[IN]` and won't read any data back, so we don't need to memcpy from host to guest.

Reply:
Works without this 👍

Reply:
Good catch Anthony!